### PR TITLE
COBRA-2015 - Neptune Broker Integration

### PR DESCRIPTION
**Pull request by sbeck14**

### CODE REVIEW

File: service/neptune.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	req, err := http.NewRequest("GET", "http://"+os.Getenv("NEPTUNE_BROKER_URL")+"/v1/neptune/url/"+servicename, nil)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		utils.ReportError(errors.New(resp.Status), r)
```
[the remark this answers is not in the document]
Comment (trevorlinton) [reply service/neptune.go]: Perhaps change the return values to `(map[string]interface{}, error)` then return the error object and nil (or an empty `map[string]interface{}` if it cannot be completed.

[the remark this answers is not in the document]
Comment (sbeck14) [reply service/neptune.go]: oops - took that out and then forgot to commit the change. I changed it to return `nil, errors.New(resp.Status)` as suggested, and then made the changes to service/configvars.go. will commit in a bit. thanks!

File: service/neptune.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	}
+
+	var brokerresponse structs.Brokerresponse
+	if resp.StatusCode == 200 || resp.StatusCode == 201 {
+		bb, _ := ioutil.ReadAll(resp.Body)
+		_ = json.Unmarshal(bb, &brokerresponse)
+		message.Status = 201
+		message.Message = brokerresponse.Response
+		r.JSON(message.Status, message)
+	}
```
Comment (trevorlinton) [inline service/neptune.go]: What happens if resp.StatusCode isnt 200 or 201?

File: service/neptune.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+		utils.ReportError(err, r)
+		return
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		utils.ReportError(errors.New(resp.Status), r)
+		return
+	}
```
Comment (trevorlinton) [inline service/neptune.go]: If resp.StatusCode != 200 condition fails it errors, but below it seems to allow resp.StatusCode == 201 (a condition never hit).

**Pull request merged**